src/main/java/com/novi/TechItEasy/service/TelevisionService.java

```diff
@@ -0,0 +1,153 @@
+        Optional<Television> television = televisionRepo.findById(televisionId);
+        Optional<RemoteControl> remoteControl = remoteControlRepository.findById(remoteControlId);
+        if (television.isPresent() && remoteControl.isPresent()) {
+            Television tele = television.get();
+            RemoteControl remoteControl1 = remoteControl.get();
+            tele.setRemoteControl(remoteControl1);
+            televisionRepo.save(tele);
+        } else {
+            throw new RecordNotFoundException("Television or Remote Controller Not Found. Try again with new Id");
+        }
+    }
+
+    public void assignCIModuleToTelevison(Long televisonId, Long ciModuleId) {
+        Optional<Television> television = televisionRepo.findById(televisonId);
+        Optional<CiModule> ciModule = ciModuleRepository.findById(ciModuleId);
+        if (television.isPresent() && ciModule.isPresent()) {
+            Television television1 = television.get();
+            television1.setCiModule(ciModule.get());
+            televisionRepo.save(television1);
+        } else {
+            throw new RecordNotFoundException("Television or CI module not found");
+        }
+    }
+
+
+//    ----------------CONVERSION---------------
+
+    // CREATE convertTelevisionToTelevisionDto() METHOD THAT RETURNS TELEVISIONDTO
+    private TelevisionDto convertTelevisionToTelevisionDto(Television televisionData) {
+
+//        CREATE INSTANCE OF OUR TELEVISONDTO CLASS;
+        TelevisionDto televisionDto = new TelevisionDto();
+
+//        SET PROPERTIES IN DTO FROM televisionData;
+        televisionDto.setId(televisionData.getId());
+        televisionDto.setBluetooth(televisionData.getBluetooth());
+        televisionDto.setAvailableSize(televisionData.getAvailableSize());
+        televisionDto.setAmbiLight(televisionData.getAmbiLight());
+        televisionDto.setBrand(televisionData.getBrand());
+        televisionDto.setHdr(televisionData.getHdr());
+        televisionDto.setName(televisionData.getName());
+        televisionDto.setPrice(televisionData.getPrice());
+        televisionDto.setRefreshRate(televisionData.getRefreshRate());
+        televisionDto.setOriginalStock(televisionData.getOriginalStock());
+        televisionDto.setScreenQuality(televisionData.getScreenQuality());
+        televisionDto.setSold(televisionData.getSold());
+        televisionDto.setType(televisionData.getType());
+        televisionDto.setWifi(televisionData.getWifi());
+        televisionDto.setSmartTv(televisionData.getSmartTv());
+        televisionDto.setScreenType(televisionData.getScreenType());
+        televisionDto.setVoiceControl(televisionData.getVoiceControl());
+
+        return televisionDto;
+    }
+
+    private Television convertTelevisionDtoToTelevision(TelevisionDto televisiondto) {
+        Television newtelevision = new Television();
+        newtelevision.setId(televisiondto.getId());
```

Comment:
You don't need to set the ID in this conversion method, because the ID is auto-generated, it gets set when you call the .save() method